### ISSUE TITLE
How to set timeout for flushOnTXEnd=off if no aync

### ISSUE THREAD

**Issue by HCideal** [rsyslog/rsyslog#5452]
### Expected behavior
when set flushOnTXEnd for zlibw, there has a method to set timeout to flush strm;

### Actual behavior
when set flushOnTXEnd for zlibw, thers has no method to set timeout to flush strm

### Environment
- rsyslog version: 
- platform: 
- for configuration questions/issues, include rsyslog.conf and included config files

**Comment by davidelang** [comment]
@rgerhards I agree, enabling a filter like zip or encryption should imply async for things like this to ensure that the data gets flushed to disk in a reasonable timeframe.
